apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs

```diff
@@ -116,25 +105,46 @@ private async Task TestBindNonNull<T>(SpannerDbType parameterType, T value, Func
             {
                 Assert.Equal(value, valueRead);
             }
+
+            Assert.False(await reader.ReadAsync());
         }
 
+        private async Task<SpannerDataReader> GetBindingTestReader(SpannerConnection connection, SpannerDbType parameterType, object value)
+        {
+            using var cmd = connection.CreateSelectCommand("SELECT @v");
+            cmd.Parameters.Add("v", parameterType, value);
+            return await cmd.ExecuteReaderAsync();
+        }
+
+        internal static async Task TestBindNonNull<T>(SpannerDbType parameterType, T value, SpannerConnection connection, Func<SpannerConnection, SpannerDbType, object, Task<SpannerDataReader>> getReader, Func<SpannerDataReader, T> typeSpecificReader = null)
```

Comment:
I think it *might* be simpler (particularly in the light of the above) to keep the full implementation instead of trying to generalize. It's a very small amount of shared code, but avoiding duplication has introduced more complexity IMO. If you put the GetBindingTestReader code back into here *and* into TestBindNonNull, and just copy what you need in the Postgres tests, we don't need to worry about disposal etc.

I think I'd need to work with the code in VS to see though... I'm happy for it to go in as-is.